### ISSUE TITLE
SwiftLintPlugin does not honor `included` configuration. Issue? Or expected?

### ISSUE THREAD

**Issue by mstanzianoinsulet** [realm/SwiftLint#5515]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I've recently switched to using the SwiftLint Build Tool Plugin and noticed that it does not appear to honor the `included` configuration within `.swiftlint.yml`. I have 1 `.swiftlint.yml` in the main App target directory, and it includes other directories that live at the same level - referenced as shown in the example below.

Is this expected to work? Or is it a limitation of using the Build Tool Plugin?

### Environment
* 0.54.0, and commit `d9c68882877d85901188cc6153a8b3651a014f8f`

```yml
included:
 - ../OtherDirectory
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No.
* Which Xcode version are you using (check `xcodebuild -version`)?
15.0

**Comment by SimplyDanny** [comment]
I think that configuration files cannot be read from places outside of the project structure due to Xcode executing plugins in a sandboxed environment.

But I'm not expert here, admittedly. @garricn, @tinder-cfuller: Can you help out?

**Comment by tinder-cfuller** [comment]
In this case, `included` is for including Swift files, not configuration files. @SimplyDanny is correct though, the plugin relies on the `inputFiles` it is provided and passes those as arguments to the script invocation.

https://github.com/garricn/SwiftLint/blob/9c216baf2094668cb1db834a73348d84cbe2f02c/Plugins/SwiftLintPlugin/SwiftLintPlugin.swift#L19

https://github.com/garricn/SwiftLint/blob/9c216baf2094668cb1db834a73348d84cbe2f02c/Plugins/SwiftLintPlugin/SwiftLintPlugin.swift#L40

And it makes sense that when paths are passed as arguments to the script, it then ignores the `included` option.

The reason the plugin uses the `inputFiles` is that these are the files that belong to the target being built. So from the plugin perspective these are the only files that would need to be linted.

**Comment by tinder-cfuller** [comment]
@mstanzianoinsulet The next release of SwiftLint includes improvements to the build plugin and now also includes a command plugin too. The command plugin could be used for your setup since I see no reason why it would not respect the `include` option (it is not provided with `inputFiles` for example).

[#4758 - Use BUILD_WORKSPACE_DIRECTORY in SwiftLintPlugin](https://github.com/realm/SwiftLint/pull/4758)
[#5497 - Introduce SwiftLintCommandPlugin](https://github.com/realm/SwiftLint/pull/5497)

**Comment by tinder-cfuller** [comment]
@mstanzianoinsulet The next release of SwiftLint includes improvements to the build plugin and now also includes a command plugin too. The command plugin could be used for your setup since I see no reason why it would not respect the `included` option (it is not provided with `inputFiles` for example).

[#4758 - Use BUILD_WORKSPACE_DIRECTORY in SwiftLintPlugin](https://github.com/realm/SwiftLint/pull/4758)
[#5497 - Introduce SwiftLintCommandPlugin](https://github.com/realm/SwiftLint/pull/5497)
